### PR TITLE
Adds CollectionViewSource to sort Package Source ComboBox in PM UI

### LINKED ISSUE #11298: Sort Package Source dropdown list using locale/culture setttings in PM UI
It seems package source dropdown list sorting is using a different non-locale logic to sort package sources:

image

### Reference:

https://docs.microsoft.com/globalization/locale/sorting-and-string-comparison#how-do-i-verify-that-sorting-works-correctly-in-my-code

Edit:

Related: https://developercommunity.visualstudio.com/t/Changing-order-of-Nuget-Packages-in-Opti/1532219

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Home/issues/11298

Regression? Last working version:

## Description

This PR makes package sources to be sorted according to UI locale/culture settings. 

- Adds a CollectionViewSource to PackageManagerTopPanel to sort packages sources combo box.
- Sets CollectionViewSource  Culture to current UI culture in code behind.
- Renames SourceRepoList to PackageSources where applicable.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - ~[ ] Automated tests added~
  - [x] Test exception: Manually tested. See images below
  - ~[ ] N/A~

- **Documentation**
  - ~[ ] Documentation PR or issue filled~
  - [x] N/A: UI change

### Validation

Package source orting in French: now it respects correct sorting, according to https://docs.microsoft.com/globalization/locale/sorting-and-string-comparison#how-do-i-verify-that-sorting-works-correctly-in-my-code

#### Before

image

#### After

image

### Accessiblity Insights FastPass validation

image

image

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs [hunk 1]
```diff
@@ -741,13 +741,13 @@ private async Task InitSourceRepoListAsync(UserSettings settings, CancellationTo
         private ValueTask PopulateSourceRepoListAsync(IReadOnlyCollection<PackageSourceMoniker> packageSourceMonikers, string optionalSelectSourceName, CancellationToken cancellationToken)
```
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs]: nit: Let's rename this one to be specific to collection name.

```suggestion
        private ValueTask PopulatePackageSourcesAsync(IReadOnlyCollection<PackageSourceMoniker> packageSourceMonikers, string optionalSelectSourceName, CancellationToken cancellationToken)
```

**Review by donnie-msft — APPROVED** [review]
I'm good with doing this now considering the ordering arrow-buttons don't do anything, anyway.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs (hunk 1 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs]: Renamed.

**Comment by zivkan** [comment]
Why are the linked issue and describe changes in the checklist crossed out?

Also, given this is a customer impacting change, I think it should have an issue in Home, so that the release notes mentions this improvement.

**Comment by dominoFire** [comment]
@zivkan 

> Why are the linked issue and describe changes in the checklist crossed out?

No reason at all. Just wanted to mark them as done.

> Also, given this is a customer impacting change, I think it should have an issue in Home, so that the release notes mentions this improvement.

I'm investigating if I can transfer the issue.

**Pull request merged**